skfem/visuals/matplotlib.py

```diff
@@ -137,22 +137,8 @@ def plot_meshline(m: MeshLine, z: ndarray, **kwargs):
     else:
         ax = kwargs["ax"]
 
-    xs = []
-    ys = []
-
     color = kwargs["color"] if "color" in kwargs else 'ko-'
-    for y1, y2, s, t in zip(z[m.t[0]],
```

Comment:
I don't know if this is needed though? Isn't ax.plot(m.p[0], z) enough? This could be just some copypasta from draw_mesh2d where appending None gives faster performance.

Reply:
Oh I see, this is because of the arbitrary order of m.p[0]. I think np.argsort(m.p[0]) is needed before plotting.

Reply:
Oh, when i read the old `for`-loop implementation, I thought it was deliberately drawing one line segment per cell, possibly so that this would work with 
* unsorted `.p` or
* discontinuous Galerkin?

Reply:
I opened #676 to discuss whether `plot_meshline` needs revision.

Reply:
> I don't know if this is needed though? Isn't `ax.plot(m.p[0], z)` enough?

It is indeed, or even using `basis.doflocs` and plotting all the ordinates, which is richer if using `ElementLineP2`.  I've gone with that in bdc010b and reverted the changes to `plot(MeshLine1)`.

That might be revisited in #676, or perhaps the resolution to that will end of being again, &lsquo;isn't `ax.plot(m.p[0], z)` enough?&rsquo;